roles/os_host_monitoring/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+#example run:
+# ansible-playbook -e "cli_environment=ops"  -e "zagg_verify_ssl=True"                        \
+#                  -e "cli_zagg_url=https://oso-rhel7-zagg-web.v6r2.ops.openshiftapps.com"    \
+#                  deploy_systemd_monitoring_containers.yml
+# FIXME: we need to change "cli_environment" to "clusterid" as that's what it really is now.
+#
+
+#  vars_files:
+#    - private_vars/monitoring_containers/zagg-server-vars.yml
+#    - private_vars/monitoring_containers/docker-registry.ops.yml
+
+- fail:
+    msg: "This playbook requires {{item}} to be set."
+  when: "{{ item }} is not defined or {{ item }} == ''"
+  with_items:
+  - oshm_zagg_web_url
+  - oshm_host_monitoring
+  - oshm_zagg_client
+  - oshm_docker_registry_url
+  - oshm_default_zagg_server_user
+  - oshm_default_zagg_server_password
+
+- name: create /etc/docker/ops
+  file:
+    path: /etc/docker/ops
+    state: directory
+    mode: 0770
+    group: root
+    owner: root
+
+- name: Copy dockercfg to /etc/docker/ops
+  template:
+      src: docker-registry.ops.cfg.j2
+      dest: /etc/docker/ops/.dockercfg
+      owner: root
+      group: root
+      mode: 0600
+
+- name: "Copy {{ oshm_host_monitoring }} systemd file"
+  template:
+    src: "{{ oshm_host_monitoring }}.service.j2"
+    dest: "/etc/systemd/system/{{ oshm_host_monitoring }}.service"
+    owner: root
+    group: root
+    mode: 0644
+  register: systemd_host_monitoring
+
+- name: "Copy {{ oshm_zagg_client }} systemd file"
+  template:
+    src: "{{ oshm_zagg_client }}.service.j2"
+    dest: "/etc/systemd/system/{{ oshm_zagg_client }}.service"
+    owner: root
+    group: root
+    mode: 0644
+  register: zagg_systemd
+
+- name: reload systemd
+  command: /usr/bin/systemctl --system daemon-reload
+  when: systemd_host_monitoring.changed or zagg_systemd.changed
+
+- name: pause for a few seconds
+  pause: seconds=2
+
+- name: "Pull docker {{ oshm_host_monitoring }}"
+  command: "docker pull {{ oshm_docker_registry_url }}{{ oshm_host_monitoring }}"
+  environment:
+    HOME: /etc/docker/ops
+
+- name: "Pull docker {{ oshm_zagg_client }}"
+  command: "docker pull {{ oshm_docker_registry_url }}{{ oshm_zagg_client }}"
+  environment:
+    HOME: /etc/docker/ops
+
+- name: "Restart the {{ oshm_host_monitoring }} service"
+  service:
+    name: "{{ oshm_host_monitoring }}"
+    state: restarted
+    enabled: yes
```

Comment:
This is not idempotent. We should do this as a notify when laying down the systemd unit files.